e2e/fee_middleware_test.go

```diff
@@ -611,6 +611,150 @@ func (s *FeeMiddlewareTestSuite) TestPayPacketFeeAsync_SingleSender_NoCounterPar
+		s.Require().NoError(err)
+		s.Require().NoError(chainATx.Validate(), "chain-a ibc transfer tx is invalid")
+	})
+
+	t.Run("tokens are escrowed", func(t *testing.T) {
+		actualBalance, err := s.GetChainANativeBalance(ctx, chainAWallet1)
+		s.Require().NoError(err)
+
+		expected := testvalues.StartingTokenAmount - walletAmount1.Amount
+		s.Require().Equal(expected, actualBalance)
+	})
+
+	t.Run("pay packet fee", func(t *testing.T) {
+		t.Run("no incentivized packets", func(t *testing.T) {
+			packets, err := s.QueryIncentivizedPacketsForChannel(ctx, chainA, channelA.PortID, channelA.ChannelID)
+			s.Require().NoError(err)
+			s.Require().Empty(packets)
+		})
+
+		packetId := channeltypes.NewPacketID(channelA.PortID, channelA.ChannelID, chainATx.Packet.Sequence)
+		packetFee1 := feetypes.NewPacketFee(testFee, chainAWallet1.Bech32Address(chainA.Config().Bech32Prefix), nil)
+		packetFee2 := feetypes.NewPacketFee(testFee, chainAWallet2.Bech32Address(chainA.Config().Bech32Prefix), nil)
+
+		t.Run("paying packetFee1 should succeed", func(t *testing.T) {
+			payPacketFeeTxResp, err = s.PayPacketFeeAsync(ctx, chainA, chainAWallet1, packetId, packetFee1)
+			s.Require().NoError(err)
+			s.AssertValidTxResponse(payPacketFeeTxResp)
+		})
+		t.Run("paying packetFee2 should succeed", func(t *testing.T) {
+			payPacketFeeTxResp, err = s.PayPacketFeeAsync(ctx, chainA, chainAWallet2, packetId, packetFee2)
+			s.Require().NoError(err)
+			s.AssertValidTxResponse(payPacketFeeTxResp)
+		})
+
+		t.Run("there should be incentivized packets", func(t *testing.T) {
+			packets, err := s.QueryIncentivizedPacketsForChannel(ctx, chainA, channelA.PortID, channelA.ChannelID)
+			s.Require().NoError(err)
+			s.Require().Len(packets, 1)
+			actualFee1 := packets[0].PacketFees[0].Fee
+			actualFee2 := packets[0].PacketFees[1].Fee
+
+			s.Require().True(actualFee1.RecvFee.IsEqual(testFee.RecvFee))
+			s.Require().True(actualFee1.AckFee.IsEqual(testFee.AckFee))
+			s.Require().True(actualFee1.TimeoutFee.IsEqual(testFee.TimeoutFee))
+
+			s.Require().True(actualFee2.RecvFee.IsEqual(testFee.RecvFee))
+			s.Require().True(actualFee2.AckFee.IsEqual(testFee.AckFee))
+			s.Require().True(actualFee2.TimeoutFee.IsEqual(testFee.TimeoutFee))
+		})
+
+		t.Run("balance should be lowered by sum of recv ack and timeout", func(t *testing.T) {
+			// The balance should be lowered by the sum of the recv, ack and timeout fees.
+			actualBalance1, err := s.GetChainANativeBalance(ctx, chainAWallet1)
+			s.Require().NoError(err)
+
+			expected1 := testvalues.StartingTokenAmount - walletAmount1.Amount - testFee.Total().AmountOf(chainADenom).Int64()
+			s.Require().Equal(expected1, actualBalance1)
+
+			actualBalance2, err := s.GetChainANativeBalance(ctx, chainAWallet2)
+			s.Require().NoError(err)
+
+			expected2 := testvalues.StartingTokenAmount - testFee.Total().AmountOf(chainADenom).Int64()
+			s.Require().Equal(expected2, actualBalance2)
```

Comment:
it might be worth splitting these up into two separate subtests, in the output of a test failure we would need to rely on line number to know which balance was incorrect.